software/firmware/firmware.c

```diff
@@ -257,12 +268,20 @@ int main()
    }
    printf("\n");
 #else
-   printf("42e61e174fbb3897d6dd6cef3dd2802fe67b331953b06114a65c772859dfc1aa\n");
+   printf("2312394bd99545d9de131c24efb781e765ac1aec243f2ed9347597a793a415e9\n");
```

Comment:
You could get the sha256 digest from automatically instead of hardcoding for each test.
Write the test input to a file: `<test_file>.bin`
run the linux command `sha256sum <test_file>.bin`, which gives the digest for the corresponding input